wordpress-mobile/WordPress-iOS
# Login: remove signin inheritance

Refs #7272

It's a good time to nuke the inheritance of `Signin*` classes for all new `Login*` classes.

To test:
- make sure logins work. Touched a lot here, tho hopefully my cut & paste skills were good.

Needs review: @aerych 
Or I can send it to someone more optional-friendly.
